src/app/ReadClient.h

```diff
@@ -59,17 +60,26 @@ class ReadClient : public Messaging::ExchangeDelegate
     {
     public:
         virtual ~Callback() = default;
+
         /**
-         * Notification that a list of events is received on the given read client.
+         * OnResponse will be called when a report data response has been received and processed for the given path.
+         *
          * The ReadClient object MUST continue to exist after this call is completed.
          *
-         * @param[in]  apReadClient         The read client which initialized the read transaction.
-         * @param[in]  apEventReportsReader TLV reader positioned at the list that contains the events.  The
-         *                                  implementation of EventStreamReceived is expected to call Next() on the reader to
-         *                                  advance it to the first element of the list, then process the elements from beginning to
-         *                                  the end. The callee is expected to consume all events.
+         * This callback will be called when:
+         *   - Receiving event data as response of Read interactions
+         *   - Receiving event data as reports of subscriptions
+         *   - Receiving event data as initial reports of subscriptions
```

Comment:
These three sentences can be compressed down to just  "called when receiving event data received in the Read and Subscribe interactions".

Reply:
updated. thanks